api/src/main/java/org/eclipse/microprofile/metrics/MetricRegistry.java

```diff
@@ -260,88 +431,95 @@ public static String name(Class<?> klass, String... names) {
     public abstract SortedSet<String> getNames();
 
     /**
-     * Returns a map of all the gauges in the registry and their names.
+     * Returns a set of the {@link MetricID}s of all the metrics in the registry.
+     *
+     * @return the MetricIDs of all the metrics
+     */
+    public abstract SortedSet<MetricID> getMetricIDs();
+
+    /**
+     * Returns a map of all the gauges in the registry and their {@link MetricID}s.
      *
      * @return all the gauges in the registry
      */
-    public abstract SortedMap<String, Gauge> getGauges();
+    public abstract SortedMap<MetricID, Gauge> getGauges();
```

Comment:
If `MetricID` is to be used as keys for a sorted map, then it must implement `java.lang.Comparable`

Reply:
ok, fixed